daft/connect/__init__.py

```diff
@@ -0,0 +1,115 @@
+"""Daft Connect provides a simple way to start a Spark server and connect to it from a client.
+
+Additionally, Daft Connect provides a Ray remote class that wraps the the daft.daft.connect_start function.
+
+Usage:
+```py
+from daft.daft import connect_start
+
+connect_server = connect_start()  # optionally pass in a port `connect_start(port=1234)`
+
+# Get the url to connect to the spark service
+port = connect_server.port()
+url = f"sc://localhost:{port}"
+
+# Connect to the spark service
+spark = SparkSession.builder.remote(url).getOrCreate()
+
+spark.createDataFrame([("hello", 1), ("world", 2)], ["word", "count"]).show()
+
+# Shutdown the server
+connect_server.shutdown()
+```
+
+Ray Usage:
+
+
+For ray, we provide an Actor class that can be used to start a daft connect server on your ray cluster.
+
+
+```py
+import ray
+from daft.connect import DaftConnectRayAdaptor
+from pyspark.sql import SparkSession
+
+# we recommend using the detached lifetime for the server
+# this will keep the server running even if the client disconnects
+server = DaftConnectRayAdaptor.options(lifetime="detached").remote()
+
+# Get the url to connect to the spark service
+url = ray.get(server.spark_remote_url.remote())
```

Comment:
Given how similar the UX is here, what do you think of something like this?

```python
connect_server = daft.spark.connect_start_ray(
    lifetime="detached",
    run_on_head_node=True,
)
url = connect_server.spark_remote_url()
```

We can just throw an error inside of `connect_start_ray` if it can't find an active Ray connection (i.e. the user hasn't called `ray.init()` yet)

Could be the same UX as the local one as well!

Reply:
I wanted to just use a ray function, but for some reason I was having some serialization issues with a function and had to use an actor instead.

Reply:
> We can just throw an error inside of connect_start_ray if it can't find an active Ray connection (i.e. the user hasn't called ray.init() yet)

doesn't our `set_runner_ray` already start one if one doesnt exist though?